dashboard-api/internal/platform/server/handler/studies/list.go

```diff
@@ -0,0 +1,113 @@
+package studies
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+)
+
+type studyReponse struct {
+	Id        string `json:"id"`
+	Name      string `json:"name"`
+	Slug      string `json:"slug"`
+	CreatedAt int64  `json:"createdAt"`
+}
+
+func ListHandler(studyRepository studiesmanager.StudyRepository) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		pagination, err := parsePaginationQueryParams(ctx)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		studies, err := studyRepository.GetStudies(ctx, pagination.Cursor, pagination.Number)
+		if err != nil {
+			ctx.Status(http.StatusInternalServerError)
+			return
+		}
+
+		data := make([]studyReponse, 0, len(studies))
+		for _, study := range studies {
+			data = append(data, studyReponse{
+				Id:        study.ID(),
+				Name:      study.Name(),
+				Slug:      study.Slug(),
+				CreatedAt: study.CreatedAt(),
+			})
+		}
+
+		if len(studies) < pagination.Number {
+			ctx.JSON(http.StatusOK, gin.H{
+				"data": data,
+				"pagination": gin.H{
+					"nextCursor": nil,
+				},
+			})
+		} else {
+			ctx.JSON(http.StatusOK, gin.H{
+				"data": data,
+				"pagination": gin.H{
+					"nextCursor": createNextCursor(pagination),
+				},
+			})
+		}
+	}
+}
+
+type paginationQueryParams struct {
+	CursorInBase64url string `form:"cursor,default=MA==" binding:"base64url"`
+	Number            int    `form:"number,default=20" binding:"gte=1,lte=100"`
+}
+
+type pagination struct {
+	Cursor int
+	Number int
+}
+
+func parsePaginationQueryParams(ctx *gin.Context) (pagination, error) {
+	var params paginationQueryParams
+
+	if err := ctx.ShouldBindQuery(&params); err != nil {
+		return pagination{}, err
+	}
+
+	cursor, err := parseCursor(params.CursorInBase64url)
+	if err != nil {
+		return pagination{}, err
+	}
+
+	return pagination{
+		Cursor: cursor,
+		Number: params.Number,
+	}, nil
+}
+
+func parseCursor(cursorInBase64url string) (int, error) {
+	decodedCursorInBytes, err := base64.URLEncoding.DecodeString(cursorInBase64url)
+	if err != nil {
+		return 0, fmt.Errorf("invalid cursor: %s", cursorInBase64url)
+
+	}
+
+	cursor, err := strconv.Atoi(string(decodedCursorInBytes))
+	if err != nil {
+		return 0, fmt.Errorf("invalid cursor: %s", cursorInBase64url)
+	}
+
+	if cursor < 0 {
+		return 0, fmt.Errorf("invalid cursor: %s", cursorInBase64url)
+	}
+
+	return cursor, nil
+}
+
+func createNextCursor(pagination pagination) string {
+	nextCursorAsString := strconv.Itoa(pagination.Cursor + pagination.Number)
+
+	return base64.URLEncoding.EncodeToString([]byte(nextCursorAsString))
```

Comment:
What's the point of base64 encoding an integer?

Reply:
In general it's good practice for the pagination cursor to be an opaque string.

This is specially helpful for public api's that have many third-party consumers.

By providing an opaque string that should not be parsed by the api consumers, the owners of the public api are able to change the underlying implementation of the pagination with more chance of avoiding unexpected breaking changes in their consumers applications.

Having said that, this api will only have a consumer for now and it's a first party application, so I'm ok if you prefer to just pass the integer and avoid this complexity for now.

Let me know what you prefer.